### PR TITLE
Support Default Credentail Provider

### PR DESCRIPTION
**Pull request by mavanidipen365**

### CODE REVIEW

File: runtime/src/main/java/io/quarkiverse/logging/cloudwatch/auth/CloudWatchCredentialsProvider.java [hunk 1]
```diff
@@ -1,4 +1,4 @@
-/*
+//*
```
Comment (bennetelli) [inline runtime/src/main/java/io/quarkiverse/logging/cloudwatch/auth/CloudWatchCredentialsProvider.java:1]: please remove this additional /

**Comment by bennetelli** [comment]
Please format both files. We are getting a failed build with this message:
Error:  Failed to execute goal net.revelc.code.formatter:formatter-maven-plugin:2.23.0:validate (default-cli) on project quarkus-logging-cloudwatch: File '/home/runner/work/quarkus-logging-cloudwatch/quarkus-logging-cloudwatch/runtime/src/main/java/io/quarkiverse/logging/cloudwatch/auth/CloudWatchCredentialsProvider.java' has not been previously formatted. Please format file (for example by invoking `mvn -f runtime net.revelc.code.formatter:formatter-maven-plugin:2.23.0:format`) and commit before running validation! -> [Help 1]

**Review by bennetelli — CHANGES_REQUESTED** [review]

File: runtime/src/main/java/io/quarkiverse/logging/cloudwatch/LoggingCloudWatchConfig.java [hunk 2]
```diff
@@ -109,19 +109,29 @@ public class LoggingCloudWatchConfig {
     @ConfigItem
     public Optional<Duration> apiCallTimeout;
 
+    /**
+     * Default credentials provider enabled added as a {@code credentials.provider.type.default}
+     */
+    @ConfigItem(name = "credentials.provider.type.default", defaultValue = "false")
```
Comment (gwenneg) [inline runtime/src/main/java/io/quarkiverse/logging/cloudwatch/LoggingCloudWatchConfig.java:115]: Let's match the usual naming convention for config properties:
```suggestion
    @ConfigItem(name = "quarkus.log.cloudwatch.default-credentials-provider.enabled", defaultValue = "false")
```

**Review by gwenneg — COMMENTED** [review]
Thank you for your contribution @mavanidipen365.

Could you please add a new entry in [config.adoc](https://github.com/quarkiverse/quarkus-logging-cloudwatch/blob/main/docs/modules/ROOT/pages/config.adoc) to document the new config property?

**Comment by mavanidipen365** [comment]
Hi @bennetelli and @gwenneg requested changes are incorporated now

**Comment by bennetelli** [comment]
> Hi @bennetelli and @gwenneg requested changes are incorporated now

Please run: mvn -f runtime net.revelc.code.formatter:formatter-maven-plugin:2.23.0:format
So the build isn't failing anymore. I am not allowed to format it.

**Comment by mavanidipen365** [comment]
> > Hi @bennetelli and @gwenneg requested changes are incorporated now
> 
> Please run: mvn -f runtime net.revelc.code.formatter:formatter-maven-plugin:2.23.0:format So the build isn't failing anymore. I am not allowed to format it.

I've applied the formatter and also given access to repo

**Review by bennetelli — CHANGES_REQUESTED** [review]

File: runtime/src/main/java/io/quarkiverse/logging/cloudwatch/LoggingCloudWatchConfig.java [hunk 3]
```diff
@@ -109,18 +109,26 @@ public class LoggingCloudWatchConfig {
     @ConfigItem
     public Optional<Duration> apiCallTimeout;
 
+    /**
+     * Default credentials provider enabled added as a {@code quarkus.log.cloudwatch.default-credentials-provider.enabled}
+     */
+    @ConfigItem(name = "quarkus.log.cloudwatch.default-credentials-provider.enabled", defaultValue = "false")
```
[the remark this answers is not in the document]
Comment (mavanidipen365) [reply runtime/src/main/java/io/quarkiverse/logging/cloudwatch/LoggingCloudWatchConfig.java:115]: I've tested on my setup, we don't needed to add prefix

[the remark this answers is not in the document]
Comment (gwenneg) [reply runtime/src/main/java/io/quarkiverse/logging/cloudwatch/LoggingCloudWatchConfig.java:115]: My bad, `default-credentials-provider.enabled` should be enough indeed.

[the remark this answers is not in the document]
Comment (mavanidipen365) [reply runtime/src/main/java/io/quarkiverse/logging/cloudwatch/LoggingCloudWatchConfig.java]: @gwenneg you are right only default-credentials-provider.enabled is needed
I've tested by removing prefix
please see https://github.com/quarkiverse/quarkus-logging-cloudwatch/pull/441/commits/f07379eec5bdfb251054833d89ebece4930518c3

[the remark this answers is not in the document]
Comment (mavanidipen365) [reply runtime/src/main/java/io/quarkiverse/logging/cloudwatch/LoggingCloudWatchConfig.java]: Hi @gwenneg and @bennetelli 
Does any pending changes required by me?

**Review by bennetelli — APPROVED** [review]

**Pull request merged**